projects/vanilla/src/app/configurators/app-configuration/global-configurator.component.ts

```diff
@@ -92,25 +116,66 @@ export class GlobalConfiguratorComponent implements OnInit {
   primary?;
   secondary?;
 
+  private subscription: Subscription;
+
+  constructor(private appService: AppService,
+    private configService: ConfigService) { }
+
   ngOnInit(): void {
     this.background = getComputedStyle(document.body).getPropertyValue('--background-color').trim();
     this.gradient = getComputedStyle(document.body).getPropertyValue('--gradient-color').trim();
 
     this.brand = getComputedStyle(document.body).getPropertyValue('--brand-300').trim();
     this.primary = getComputedStyle(document.body).getPropertyValue('--primary-300').trim();
     this.secondary = getComputedStyle(document.body).getPropertyValue('--secondary-300').trim();
+
+    this.setupHighlights();
+  }
+
+  ngOnDestroy(): void {
+    if (this.subscription) this.subscription.unsubscribe();
   }
 
   setTextColor(index: number) {
-    this.context.config.textColor = `text${index+1}`;
+    this.context.config.textColor = `text${index + 1}`;
     this.context.configChanged();
   }
 
   changeTheme(event) {
-    const checked =  event.target.checked;
+    const checked = event.target.checked;
     this.context.config.theme = checked;
     this.context.configChanged();
 
     document.body.classList.toggle("sinequa");
   }
+
+  enableHighlight(highlight: PreviewHighlightColors) {
+    if (!highlight['$enabled']) {
+      highlight.bgColor = undefined;
+      highlight.color = undefined;
+    }
+    this.context.configChanged();
+  }
+
+  private setupHighlights() {
+    this.subscription = this.configService.watchConfig("global")
+      .subscribe(value => {
+        if (!this.context.config.entityHighlights) {
+          this.context.config.entityHighlights = [...(value.entityHighlights || [])].map(h => h['$enabled'] = true);
```

Comment:
Je pense que cette ligne ne marche pas car la fonction map() retourne un array de true.

Reply:
Plus généralement j'ai du mal à comprendre la logique du code: Est-ce que le `context.config` n'est pas le même objet que le `value` ici?

Reply:
Ce passage n'est censé être traversé que quand il n'y a pas encore de contexte (d'où le `if (!this.context.config.entityHighlights)` juste avant), c'est une instruction plutôt pour faire un setup la première fois, ou après un reset.

Pour la liste du map, je ne sais pas trop car en utilisant ça semble fonctionner, mais ça semble plus safe de mettre entre accolades et returner l'objet oui.

Reply:
>quand il n'y a pas encore de contexte
Mais vu que maintenant tu as mis `entityHighlights` directement dans la config par défaut, il devrait exister dès le début, non? Et par ailleurs, pourquoi est-ce que le `entityHighlights` existerait dans `value` et pas dans `context.config`?

Reply:
Ca c'est peut-être moi qui comprend pas encore tout de l'appli 😄 Je vais faire des tests et retirer ce morceau s'il ne sert à rien en effet

Reply:
@ericleib En effet c'était pas nécessaire, j'ai pu bien nettoyer en retirant le Subscription/OnDestroy